Source/Core/BingMapsGeocoderService.js

```diff
@@ -0,0 +1,113 @@
+/*global define*/
+define([
+    './BingMapsApi',
+    './defaultValue',
+    './defineProperties',
+    './loadJsonp',
+    './Rectangle',
+], function(
+    BingMapsApi,
+    defaultValue,
+    defineProperties,
+    loadJsonp,
+    Rectangle) {
+    'use strict';
+
+   var url = 'https://dev.virtualearth.net/REST/v1/Locations';
+
+    /**
+     * Provides geocoding through Bing Maps.
+     * @alias BingMapsGeocoderService
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} [key] A key to use with the Bing Maps geocoding service
+     * @param {Boolean} autoComplete Indicates whether this service shall be used to fetch auto-complete suggestions
+     */
+    function BingMapsGeocoderService(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        this._canceled = false;
+        this._url = 'https://dev.virtualearth.net/REST/v1/Locations';
+        this._key = BingMapsApi.getKey(options.key);
+
+        /**
+         * Indicates whether this geocoding service is to be used for autocomplete.
+         *
+         * @type {boolean}
+         * @default false
+         */
+        this.autoComplete = defaultValue(options.autoComplete, false);
```

Comment:
Even if a particular `GeocoderService` can technically be used for autocomplete, an app developer may want to turn it off. If we have Bing and Nominatim, for example, the Nominatim terms of service might be violated when an app uses autocomplete due to the greater number of requests, while Bing's ToS are fine with that amount. Or if we have two geocoder services, but one's autocomplete results consistently better, we might want to turn off autocomplete for the other service to avoid polluting the autocomplete results list.

In the case of `BingMapsGeocoderService` autocomplete is turned off by default because the Bing Maps API is not designed to support it well.

So my intention was that the widget-level `autoComplete` property indicates whether the widget should enable autocomplete, and the geocoder services `autoComplete` property indicates whether each service supports it.